fiftyone/core/collections.py

```diff
@@ -2031,16 +1947,85 @@ def _add_view_stage(self, stage):
         """
         raise NotImplementedError("Subclass must implement _add_view_stage()")
 
+    def aggregate(self, aggregations, _attach_frames=True):
+        """Aggregates one or more
+        :class:`fiftyone.core.aggregations.Aggregation` instances.
+
+        Note that it is best practice to group aggregations into a single call
+        to :meth:`aggregate() <aggregate>`, as this will be more efficient than
+        performing multiple aggregations in series.
+
+        Args:
+            aggregations: an :class:`fiftyone.core.aggregations.Aggregation` or
+                iterable of :class:`<fiftyone.core.aggregations.Aggregation>`
+                instances
+
+        Returns:
+            an :class:`fiftyone.core.aggregations.AggregationResult` or list of
+            :class:`fiftyone.core.aggregations.AggregationResult` instances
+            corresponding to the input aggregations
+        """
+        scalar_result, aggregations, facets = self._build_aggregation(
+            aggregations
+        )
+        if len(aggregations) == 0:
+            return []
+
+        # pylint: disable=no-member
+        pipeline = self._pipeline(
+            pipeline=facets, attach_frames=_attach_frames
+        )
+        try:
+            # pylint: disable=no-member
+            result = next(self._dataset._sample_collection.aggregate(pipeline))
+        except StopIteration:
+            pass
+
+        return self._process_aggregations(aggregations, result, scalar_result)
+
+    def _pipeline(
+        self,
+        pipeline=None,
+        attach_frames=True,
+        hide_frames=False,
+        squash_frames=False,
+    ):
+        """Returns the MongoDB aggregation pipeline for the collection.
+
+        Args:
+            pipeline (None): a MongoDB aggregation pipeline (list of dicts) to
+                append to the current pipeline
+            attach_frames (True): whether to attach the frame documents to the
+                result. Only applicable to video datasets
+            hide_frames (False): whether to hide frames in the result. Only
+                applicable to video datasets
+            squash_frames (False): whether to squash frames in the result. Only
+                applicable to video datasets
+
+        Returns:
+            the aggregation pipeline
+        """
+        raise NotImplementedError("Subclass must implement _pipeline()")
+
     def _aggregate(
-        self, pipeline=None, hide_frames=False, squash_frames=False
+        self,
```

Comment:
Apologies for the diff here; I didn't change the implementation, I just added some content to the docstrings and reorganized the methods